src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -170,7 +176,22 @@ public void cancelTask(String taskId,
         Map<String, CamundaVariable> variables = camundaService.getTaskVariables(taskId);
         roleAssignmentVerification(variables, accessControlResponse.getRoleAssignments(), permissionsRequired);
 
-        camundaService.cancelTask(taskId);
+        boolean isFeatureEnabled = launchDarklyFeatureFlagProvider.getBooleanValue(
+            FeatureFlag.RELEASE_2_CANCELLATION_COMPLETION_FEATURE,
+            accessControlResponse.getUserInfo().getUid()
+        );
+
+        if (isFeatureEnabled) {
+            //Lock & update Task
```

Comment:
it looks like this block can be moved into a method and then reuse this method in other places where we use the same code block